tensorboard/webapp/deeplink/hash.ts

```diff
@@ -23,25 +23,28 @@ const TAB = '__tab__';
 
 @Injectable()
 export class HashDeepLinker implements DeepLinkerInterface {
-  private readonly tfStorage: TfStorageElement;
+  tfStorage: DeepLinkerInterface & {
+    migrateLegacyURLScheme: () => void;
+  };
 
   constructor() {
-    this.tfStorage = document.createElement('tf-storage');
-    const tfGlobals = document.createElement('tf-globals');
+    document.createElement('tf-storage');
+    document.createElement('tf-globals');
+    this.tfStorage = (window as any)['tensorboard']['tf_storage'];
 
     // Note: `migrateLegacyURLScheme()` must be called before `setUseHash`, so
     // that tfStorage reads from the actual URL, not the fake hash for tests
     // only.
-    tfGlobals.tf_globals.setUseHash(true);
-    this.tfStorage.tf_storage.migrateLegacyURLScheme();
+    (window as any)['tensorboard']['tf_globals'].setUseHash(true);
```

Comment:
Couldn't `setUseHash` here still be renamed in theory? So you wouldn't be calling a valid function here.

(And same with `migrateLegacyURLScheme`, `getString`, `setString` below).